### PR TITLE
storage: (MySQL) Only enforce replica_preserve_commit_order setting if connected to a replica

### PR DESCRIPTION
**Pull request by rjobanp**
### Motivation

  * This PR adds a feature that has not yet been specified.

    Only enforce the `replica_preserve_commit_order` setting when creating a MySQL Source if connected to a replica server.

    The `replica_preserve_commit_order` setting on MySQL is used to ensure that when connected to a MySQL replica, the original order of transactions received from the primary is respected, to ensure data consistency between the primary and replica. This has been highlighted as a cause of data-inconsistency issues in a recent Jepsen [report](https://jepsen.io/analyses/mysql-8.0.34) on MySQL.

### Tips for reviewer

cc @nrainer-materialize - there are some additions to the `mysql-cdc` test config here that might be interesting for other test scenarios (setting up a primary-replica cluster and connecting to the replica).

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/mysql-util/src/replication.rs [hunk 1]
```diff
@@ -51,13 +51,24 @@ pub async fn ensure_gtid_consistency(conn: &mut Conn) -> Result<(), MySqlError>
...
+/// consecutive monotonically increasing transaction ids.
 pub async fn ensure_replication_commit_order(conn: &mut Conn) -> Result<(), MySqlError> {
-    // This system variable was renamed between MySQL 5.7 and 8.0
-    match verify_sys_setting(conn, "replica_preserve_commit_order", "1").await {
-        Ok(_) => Ok(()),
-        Err(_) => verify_sys_setting(conn, "slave_preserve_commit_order", "1").await,
+    // Determine if this is a replica by checking if there are any rows in the replica connection table
+    let res: u8 = conn
+        .query_first(
+            "SELECT COUNT(*) FROM performance_schema.replication_connection_configuration LIMIT 1",
```
Comment (petrosagg) [inline src/mysql-util/src/replication.rs:61]: The `LIMIT` clause is applied after the aggregation and `COUNT(*)` already returns exactly one line so it's a no-op

```suggestion
            "SELECT COUNT(*) FROM performance_schema.replication_connection_configuration",
```

**Review by petrosagg — APPROVED** [review]

**Comment by nrainer-materialize** [comment]
Thanks for keeping me in the loop!

**Comment by nrainer-materialize** [comment]
I amended the last commit with the result of `cargo fmt`.

**Review by nrainer-materialize — APPROVED** [review]

**Comment by rjobanp** [comment]
@nrainer-materialize thank you!

**Pull request merged**